v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/PipelineController.java

```diff
@@ -158,6 +142,43 @@ static PipelineResult executeSingleInstanceMigration(
     return pipeline.run();
   }
 
+  @NotNull
+  /**
+   * This list will contain the final list of tables that actually get migrated, which will be the
+   * intersection of Spanner and source tables.
+   */
+  static List<String> listSpannerTablesToMigrate(
+      Ddl ddl, ISchemaMapper schemaMapper, Set<String> tablesToMigrateSet) {
+    // This list is all Spanner tables topologically ordered.
+    List<String> orderedSpTables = ddl.getTablesOrderedByReference();
+
+    // This list will contain the final list of tables that actually get migrated, which will be the
+    // intersection of Spanner and source tables.
+    List<String> finalTablesToMigrate = new ArrayList<>();
+    for (String spTable : orderedSpTables) {
+      try {
+        String srcTable = schemaMapper.getSourceTableName("", spTable);
+        if (!tablesToMigrateSet.contains(srcTable)) {
+          continue;
```

Comment:
It might be useful to put a log before continue, it might help highlight situations where SMT schema migration had a table but one (or more) of the shards did not have it.

Reply:
added